services/jsongrpc/src/service.rs

```diff
@@ -0,0 +1,43 @@
+// clippy warning caused by the instrument macro
+#![allow(clippy::unit_arg)]
+
+use ::rpc::mayastor::{JsonRpcReply, JsonRpcRequest};
+use common::wrapper::v0::{BusGetNode, SvcError};
+use mbus_api::message_bus::v0::{MessageBus, *};
+use rpc::mayastor::json_rpc_client::JsonRpcClient;
+use snafu::ResultExt;
+
+#[derive(Clone, Default)]
+pub(super) struct JsonGrpcSvc {}
+
+/// JSON gRPC service implementation
+impl JsonGrpcSvc {
+    /// Generic JSON gRPC call issued to Mayastor using the JsonRpcClient.
+    pub(super) async fn json_grpc_call(
+        request: &JsonGrpcRequest,
+    ) -> Result<String, SvcError> {
+        let node =
+            MessageBus::get_node(&request.node)
+                .await
+                .context(BusGetNode {
+                    node: request.node.clone(),
+                })?;
+        let mut client =
+            JsonRpcClient::connect(format!("http://{}", node.grpc_endpoint))
+                .await
+                .unwrap();
+        let response: JsonRpcReply = client
+            .json_rpc_call(JsonRpcRequest {
+                method: request.method.to_string(),
+                params: request.params.to_string(),
+            })
+            .await
+            .map_err(|_| SvcError::JsonRpc {
```

Comment:
should we collect the error as well?

Reply:
Done - included the error in the mapped error.